### ISSUE TITLE
Scaling Issues On Steam Deck (docked) v4.3.1

### ISSUE THREAD

**Issue by kpervin** [cgutman/moonlight-qt#946]
When using Moonlight v4.3.1 on the Steam Deck (in this case docked mode), I experience what almost looks like an underscanned image of the host. 
![PXL_20230113_211738596 NIGHT](https://user-images.githubusercontent.com/7747148/212421162-95df58f7-7aeb-4e7e-9269-b9c3e47b1bcc.jpg)
Note the performance overlay to give you a sense of the bezels of the display. 

Seems to occur in all display modes other than Windowed, though that ends up causing blurry text. Host PC is set to 1440p.

### Basic Settings
- Any resolution/fps
- Full screen/Borderless (does not occur in Windowed)
- [x] Vsync
- [ ] Frame Pacing

### Audio Settings
- 5.1
- [x] Mute host
- [ ] Mute audio stream

### Ui Settings
- Language: auto
- GUI Display mode: Windowed
- [x] Show connection quality settings
- [x] Keep display awake

### Host Settings
- [x] Optimize game settings
- [x] Quit app on host

### Advanced Settings
- Video Decoder: auto
- Video Codec: auto
- [ ] HDR
- [ ] Unlock unsupported FPS options
- [x] Automatically find PCs
- [x] Automatically detect blocked connections

_Originally posted by @kpervin in https://github.com/moonlight-stream/moonlight-qt/issues/900#issuecomment-1382403249_

**Comment by cgutman** [comment]
My guess is that you're using Sunshine (which preserves aspect ratio when scaling) and also not using resolution that is matched to your host or client. This will lead to both letterboxing and pillarboxing as both Sunshine and Moonlight try to preserve the aspect ratio. You can correct this by either changing your host resolution to match the stream resolution or changing your stream resolution to match your client resolution.

**Comment by kpervin** [comment]
You are correct that I'm using Sunshine, but apart from that I've tried each resolution in Moonlight's settings and I cannot seem to get rid of this issue. The host machine uses a 2560x1440 monitor, while the TV is 4K. I've tried setting Moonlight's stream resolution to 1440p, 4K, or even 1080p, but to no avail.

**Comment by kftX** [comment]
> You are correct that I'm using Sunshine, but apart from that I've tried each resolution in Moonlight's settings and I cannot seem to get rid of this issue. The host machine uses a 2560x1440 monitor, while the TV is 4K. I've tried setting Moonlight's stream resolution to 1440p, 4K, or even 1080p, but to no avail.

Did you ever fix this issue? It's started happening to me on a new Windows install and nothing can seem to fix it. It was working just fine before. I just moved houses and thought it may be Wi-Fi being too slow, but after plugging in a cable for LAN, same result. I've tried all fixes I could find here to no avail.

**Comment by kpervin** [comment]
@kftX Go to the moonlight app in Game Mode, go to the Steam settings on the far right, and select "Native Resolution" instead of "Default". That seemingly fixed it for me!

**Comment by kftX** [comment]
> @kftX Go to the moonlight app in Game Mode, go to the Steam settings on the far right, and select "Native Resolution" instead of "Default". That seemingly fixed it for me!

Huh, that's weird. I've tried default, native and forcing 4k/1440p and the issue is the same every time.

**Comment by kpervin** [comment]
You're setting it via this, right?

![image](https://github.com/user-attachments/assets/5fd7bd87-bfcb-4e72-aec6-71955063f56a)
![image](https://github.com/user-attachments/assets/ee137a2e-eafe-487e-9f51-433549567e0c)

**Comment by kftX** [comment]
> @kftX You're setting it via this, right?
> 
> ![image](https://private-user-images.githubusercontent.com/7747148/361223691-5fd7bd87-bfcb-4e72-aec6-71955063f56a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1ODUwMjEsIm5iZiI6MTcyNDU4NDcyMSwicGF0aCI6Ii83NzQ3MTQ4LzM2MTIyMzY5MS01ZmQ3YmQ4Ny1iZmNiLTRlNzItYWVjNi03MTk1NTA2M2Y1NmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjVUMTExODQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTVmYmZlOGYxYmE5MmY4YWM4ZWMzMjVjMjdmNzIzMTZkYTY2MWZjMTdkODkwZDM1Njk4MDIwMDJkYjg0ODk4MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.I4mTT1DW4H3NeEqMOJRPQBuUIsliQbrzsUMj8V2UnuE) ![image](https://private-user-images.githubusercontent.com/7747148/361223849-ee137a2e-eafe-487e-9f51-433549567e0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1ODUwMjEsIm5iZiI6MTcyNDU4NDcyMSwicGF0aCI6Ii83NzQ3MTQ4LzM2MTIyMzg0OS1lZTEzN2EyZS1lYWZlLTQ4N2UtOWY1MS00MzM1NDk1NjdlMGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjVUMTExODQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2I2NzhmNmFjYmM2NGViMDdlOTNlMGNjMGQzNTg0YTFkZGVmMzFlMTQ1OTIwYjJmOTM2NjVmMzQ4ZmMyOTU5OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.P5WCwOwM4XhK0uSJsSAhe0C2eLr_Kqgq5LQ_Nqinlo0)

Yup yup. None of it helped.

I hadn't tried in Desktop Mode, but in Desktop mode it's perfect so something is definitely "fucky" about GameMode for some reason. Desktop mode will do for now, but still... this wasn't happening before. At least now I'm 80% sure it's not Moonlight.

**Comment by kpervin** [comment]
> Did you ever fix this issue? It's started happening to me on a new Windows install and nothing can seem to fix it.

Sorry, I missed you said Windows. Are we or are we not talking about SteamOS here?

**Comment by kftX** [comment]
> > Did you ever fix this issue? It's started happening to me on a new Windows install and nothing can seem to fix it.
> 
> Sorry, I missed you said Windows. Are we or are we not talking about SteamOS here?

No no, I'm using it on SteamOS, but I'm streaming a Windows machine to it that's been freshly reinstalled and I've never had a fresh install of Windows/Sushine cause these issues with Moonlight.

**Comment by eatnooM** [comment]
Weird one, but switching from the Beta to the Stable channel seems to have solved this one for me in Gaming mode. I may have had a slightly different issue as the combined letter/pillarboxing was secondary to some utterly diabolical scaling that had me thinking the video hit 1280x800 at some point in the pipeline despite being 4k on both ends.
